flow360/component/simulation/models/surface_models.py

```diff
@@ -0,0 +1,130 @@
+"""
+Contains basically only boundary conditons for now. In future we can add new models like 2D equations.
+"""
+
+from abc import ABCMeta
+from typing import List, Literal, Optional, Tuple, Union
+
+import pydantic as pd
+
+from flow360.component.simulation.framework.base_model import Flow360BaseModel
+from flow360.component.simulation.operating_condition import VelocityVectorType
+
+from .turbulence_quantities import TurbulenceQuantitiesType
+
+
+class BoundaryBase(Flow360BaseModel, metaclass=ABCMeta):
+    type: str = pd.Field()
+    entities: EntityList[Surface] = pd.Field(alias="surfaces")
+
+
+class BoundaryBaseWithTurbulenceQuantities(BoundaryBase, metaclass=ABCMeta):
+    turbulence_quantities: Optional[TurbulenceQuantitiesType] = pd.Field(None)
+
+
+class HeatFlux(Flow360BaseModel):
+    heat_flux: Union[HeatFluxType, pd.StrictStr] = pd.Field()
```

Comment:
For these single field class that takes in simple objects I was thinking adding the before model validator to pass the input to the field so that user writes:
`HeatFlux(2*u.J/u.m/u.m)` instead of
`HeatFlux(heat_flux = 2*u.J/u.m/u.m)`

Reply:
This can wait though as we have so much on the plate right now.

Reply:
yes, I was thinking the same

Reply:
actually it is easy:

```python
class _SingleAttributeModel(Flow360BaseModel):
    def __init__(self, *args, **kwargs) -> None:    
        if len(args) == 1 and 'value' not in kwargs:
            kwargs['value'] = args[0]
        super().__init__(**kwargs)
```